### PR TITLE
GitLab CI integration

### PR DESCRIPTION
**Pull request by schaefi**
The nice part of the CI system there is it's container
concept and the ease of use to run pipelines on different
containers. The changes here references the issue reported
in Issue #822

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -1,14 +1,62 @@
...
+pytest:
+  stage: test
+  image: fedora:latest
+  <<: *install_system_deps
+  script:
+    - >
+      dnf install -y --refresh
+      xorriso genisoimage python36 python34 python2
+      python2-devel python2-pip
+    - export LANG=en_US.UTF-8
```
Comment (Conan-Kudo) [inline .gitlab-ci.yml]: This should not be necessary, as Fedora uses `C.UTF-8` by default.

Comment (schaefi) [reply .gitlab-ci.yml]: It is necessary. without it py2.7 fails

Comment (Conan-Kudo) [reply .gitlab-ci.yml]: Does it work with `C.UTF-8`?

**Comment by schaefi** [comment]
> There's no package test job either?

yes I just created this as a reference for the tox calls. I'm fine with the package job.

File: .gitlab-ci.yml (hunk 1 above)
Comment (schaefi) [reply .gitlab-ci.yml]: I haven't tested this, will change it and then we wait for the result

Comment (dcermak) [reply .gitlab-ci.yml]: Looks like it does.

**Comment by Conan-Kudo** [comment]
Closing this in favor of #980 as @dcermak will incorporate these improvements there.

**Pull request closed without merging**